### ISSUE TITLE
InvalidCastException: Default parameter value for Nullable Enum?

### ISSUE THREAD

**Issue by cpboyd** [dotnet/coreclr#25010]
In a web API, I've tried the following:
```c#
public IResponse Retrieve(int id = 0, Enums.Status? status = Enums.Status.Pending)
```

This causes the following error unless `status` is specified:
```c#
InvalidCastException: Unable to cast object of type 'System.Int32' to type 'System.Nullable`1[Api.Models.Enums+Status]'
```

However, this seems to be valid:
```c#
public IResponse Retrieve(int id = 0, Enums.Status? status = null)
```

In addition, I can make it not nullable and avoid this issue.

**Comment by jkotas** [comment]
Could you please share the full stacktrace of the error? This is likely issue for https://github.com/aspnet/aspnetcore

**Comment by MichalStrehovsky** [comment]
This sounds related to #25256 but it's hard to say without a stack trace.

@cpboyd do you have a stack trace or other details that would let us repro this?

**Comment by MichalStrehovsky** [comment]
Going to assume this is the same issue as #25256 since there is a nullable enum parameter with a default value.

**Comment by cpboyd** [comment]
@MichalStrehovsky Sorry, I've been on other projects lately.
I'd say this is partially related to #25256, since I'd noticed similar behavior (in that it seems the raw value of the enum is being passed in).

Unfortunately, however, there's no clear workaround to keep the parameter nullable AND keep the default value.

Here's the full stack trace:
```
System.InvalidCastException: Unable to cast object of type 'System.Int32' to type 'System.Nullable`1[Api.Models.Enums+Status]'.
   at lambda_method(Closure , Object , Object[] )
   at Microsoft.Extensions.Internal.ObjectMethodExecutor.Execute(Object target, Object[] parameters)
   at Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeNextActionFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeInnerFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeNextResourceFilter()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeFilterPipelineAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

**Comment by cpboyd** [comment]
Seeing that the function itself seems to work in #25256, perhaps this is more specifically an issue for AspNetCore?

**Comment by MichalStrehovsky** [comment]
Based on a cursory look this *might* be the same issue as #25256. `ObjectMethodExecutor` does deal with default parameter values. Might be worth opening an issue in the ASP.NET repo, linking to #25256 as a possible cause (they are better equipped to figure out whether it's the really the same thing).

The reflection issue got moved into the Future milestone, so (as of now) we don't expect to fix it in 3.0.

**Comment by cpboyd** [comment]
For anyone stumbling on this issue, this seems to be the same as https://github.com/aspnet/AspNetCore/issues/3165, which was fixed in https://github.com/aspnet/Extensions/pull/531.
